src/search_algo.hpp

```diff
@@ -63,23 +63,19 @@
 
 void readIndexOptions(LambdaOptions & options)
 {
-    // Check that directory exists and is readable
-    //     struct stat path_stat;
-    // //     stat(indexFilePath.c_str(), &path_stat);
-    //     if (stat(indexFilePath.c_str(), &path_stat) || !S_ISDIR(path_stat.st_mode))
-    //         throw IndexException("Index directory does not exist or is not readable.\n");
-
     fake_index_file f{options.indexFileOptions};
 
-    if (options.indexFilePath.extension() == ".lba")
+    std::string                  filename(options.indexFilePath);
+    size_t                       threads = std::min<size_t>(options.threads, 4); // more than four threads is harmful
+    bio::io::transparent_istream is{filename, {.threads = threads}};
```

Comment:
```suggestion
    bio::io::transparent_istream is{filename, {.threads = 2}};
```

Can you try this for decompression?